### PR TITLE
Run PyTorch unit tests in OSS CI

### PR DESCRIPTION
**Pull request by zasdfgbnm**
The `torch_tests.py` was taken from https://github.com/pytorch/pytorch/pull/70049, and I modified the skip list. I am running these tests in our NVIDIA CI, and I find that changes in `torch.testing._internal.distributed.distributed_test` might break this test. I think it is a good idea to maintain a copy of this file in the public `torch_ucc` repository so that we can get the signal of the status of these tests from the public OSS CI.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
cc: @srinivas212 @bryanmr

File: test/torch_tests.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+DistTestCases.skip_collective["scatter"] = {"nccl", "ucc"}
+DistTestCases.skip_collective["reduce"] = {"ucc"}
+DistTestCases.skip_collective["sendrecv anysource"] = {"nccl", "ucc"}
+DistTestCases.skip_collective["cpu barrier"] = {"nccl", "ucc"}
+
+# Sets showing that something is implemented
+DistTestCases.backend_feature["gpu"] = {"nccl", "gloo", "ucc"}
+DistTestCases.backend_feature["cuda"] = {"nccl", "gloo", "ucc"}
+DistTestCases.backend_feature["ddp"] = {"nccl", "gloo", "ucc"}
+DistTestCases.backend_feature["subgroup"] = {"nccl", "gloo", "ucc"}
```
Comment (kingchc) [inline test/torch_tests.py:31]: does this mean we run tests on "nccl" and "gloo" backend as well? Wonder why we run those backends on torch-ucc repo.

Comment (bryanmr) [reply test/torch_tests.py:31]: It just means the test would run and not get skipped for those backends. The set doesn't mean that those backends are being used. It probably doesn't matter and the set could be just {"ucc"}, but there is no harm in including them as valid.

Comment (zasdfgbnm) [reply test/torch_tests.py]: I will remove mentions to other PGs

File: test/torch_tests.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+from torch.testing._internal.common_distributed import DistTestCases
+from torch.testing._internal.common_utils import (
+    run_tests,
+    sandcastle_skip_if,
+)
+
+# Sets showing that a collective isn't implemented
+DistTestCases.skip_collective["allgather_coalesced"] = {"nccl", "mpi", "ucc"}
+DistTestCases.skip_collective["gather"] = {"nccl", "ucc"}
+DistTestCases.skip_collective["scatter"] = {"nccl", "ucc"}
```
[the remark this answers is not in the document]
Comment (zasdfgbnm) [reply test/torch_tests.py]: I have removed the mention of "nccl" here. I will write follow-up PRs to update UCC skips once #71 is merged.

File: test/torch_tests.py (hunk 1 above)
Comment (zasdfgbnm) [reply test/torch_tests.py]: removed

**Pull request closed without merging**